.gitignore

```diff
@@ -2,5 +2,4 @@
 *.ipr
 *.iws
 target
-.idea
-
+.*
```

Comment:
I prefer to ignore all dot files (as most of them do not have to be tracked) and whitelist only the handful that have to. Strictly speaking this has nothing to do with the rest of the PR, yes.